src/utils/scroller.ts

```diff
@@ -0,0 +1,101 @@
+import ScrollerFactory from 'libraries/scroller';
+import globalize from 'scripts/globalize';
+
+export enum ScrollDirection {
+  RIGHT = 'right',
+  LEFT = 'left',
+}
+
+interface ScrollState {
+  scrollPos: number;
+}
+interface ScrollerSlideViewWindowProps {
+  direction: ScrollDirection;
+  scroller: ScrollerFactory | null;
+  scrollState: ScrollState;
+}
+export function scrollerItemSlideIntoView({ direction, scroller, scrollState }: ScrollerSlideViewWindowProps) {
+    if (!scroller) {
+        return;
+    }
+
+    const slider: HTMLElement = scroller.getScrollSlider();
+    const items = [...slider.children];
+
+    if (direction === ScrollDirection.LEFT) {
+        scrollToPreviousVisibleWindow(scroller, items as HTMLElement[], scrollState);
+    } else {
+        scrollToNextVisibleWindow(scroller, items as HTMLElement[], scrollState);
+    }
+}
+
+function getFirstAndLastVisible(scrollFrame: HTMLElement, items: HTMLElement[], { scrollPos: scrollPosition }: ScrollState) {
+    const isRTL = globalize.getIsRTL();
+    const localeModifier = isRTL ? -1 : 1;
+
+    const currentScrollPos = scrollPosition * localeModifier;
+    const scrollerWidth = scrollFrame.offsetWidth;
+    const itemWidth = items[0].offsetWidth;
+
+    // Rounding down here will give us the first item index which is fully visible. We want the first partially visible
+    // index so we'll subtract one.
+    const firstVisibleIndex = Math.max(Math.floor(currentScrollPos / itemWidth) - 1, 0);
+    // Rounding up will give us the last index which is at least partially visible (overflows at container end).
+    const lastVisibleIndex = Math.floor((currentScrollPos + scrollerWidth) / itemWidth);
+
+    return [firstVisibleIndex, lastVisibleIndex];
+}
+
+function scrollToNextVisibleWindow(scroller: ScrollerFactory, items: HTMLElement[], scrollState: ScrollState) {
```

Comment:
There is a decent amount of repetition between this and `scrollToPreviousVisibleWindow`. I think if we pass the scroll direction as a parameter we could handle both in a single function.

Reply:
Good call! The logic got simpler as I refined it, but I hadn't quite taken it to the point of consolidating the functions.